packages/legend-query/src/stores/QueryBuilderGraphFetchTreeUtil.ts

```diff
@@ -183,17 +189,46 @@ export const addQueryBuilderPropertyNode = (
   ];
   let parentExplorerTreeNode = explorerTreeData.nodes.get(node.parentId);
   while (
-    parentExplorerTreeNode instanceof QueryBuilderExplorerTreePropertyNodeData
+    parentExplorerTreeNode instanceof
+      QueryBuilderExplorerTreePropertyNodeData ||
+    parentExplorerTreeNode instanceof QueryBuilderExplorerTreeSubTypeNodeData
   ) {
+    let subType =
+      OptionalPackageableElementExplicitReference.create<Class>(undefined);
+    let subtypeAssigned = false;
+    while (
+      parentExplorerTreeNode instanceof QueryBuilderExplorerTreeSubTypeNodeData
+    ) {
+      if (!subtypeAssigned) {
+        subType = OptionalPackageableElementExplicitReference.create(
+          parentExplorerTreeNode.subclass,
+        );
+        subtypeAssigned = true;
+      }
+      parentExplorerTreeNode = explorerTreeData.nodes.get(
+        parentExplorerTreeNode.parentId,
+      );
+    }
+
+    guaranteeType(
```

Comment:
We shouldn't break the app because of a user input.
You could do 2 things here. 
1. disable the drag and drop on that property
2. allow drag and drop and just provide an error notification when a user does with this error message.

Would go for 2.